### ISSUE TITLE
Fix the API

### ISSUE THREAD

**Issue by jamestalmage** [jamestalmage/append-transform#3]
Right now, this is the API:

```js
var captureRequire = require('capture-require');
var transform = require('my-transform');

captureRequire(function(module, source, file) {
  module._compile(transform(source), file);
});
```

I don't like it. It suggests your allowed to do some shenanigans like replace module._compile.

I think it should be more like `pirates`, with just a simple transform:

```js
captureRequire(function(source, filename) {
  return transform(source);
});
```

I think I will avoid a filter function option though, since you can just do this:

```js
captureRequire(function(source, filename) {
  if (shouldApply(filename)) {
    source =  transform(source);
  }
  return source;
});
```

I will put something about [`node-modules-regexp`](https://www.npmjs.com/package/node-modules-regexp) in the docs for people who want to avoid `node_modules`. I just think almost anyone who uses this isn't going to be well served by a boolean `avoidNodeModules` setting (`nyc` would certainly disable that feature).

// @novemberborn, @bcoe

**Comment by novemberborn** [comment]
:+1:

**Comment by sindresorhus** [comment]
:+1:

**Comment by jamestalmage** [comment]
Just published [`append-transform@0.2.0`](https://www.npmjs.com/package/append-transform) with these changes. Thanks.
